src/hub.js

```diff
@@ -1,23 +1,64 @@
 import uuid from 'uuid';
 import { log } from './log';
 import { app } from './app';
-import { postMessage, hubMessageValid, publishMessageValid } from './msg';
-import { HEARTBEAT } from './lib';
+import { postMessage, hubMessageValid, complexMessageValid } from './msg';
+import { HEARTBEAT, CHROME_APP_ID } from './lib';
+
+let hubInstance;
 
 /**
  * Special features supplied by the Tradeshift® Chrome™.
  * @typedef {object} ChromeWindowFeatures
- * @property {function(Window): string} appIdByWindow Called to get an appId based on a Window object.
- * @property {function(string, Window): Window} windowByAppId Called to get a window object based on an appId and the requesting app's Window object.
- * @property {function(Window): void} appTimeout Called when an app fails to reply in time to a PING request.
+ * @property {function(Window): string} appByWindow Called to get an appId based on a Window object.
+ * @property {function(string, Window): Window} windowByApp Called to get a window object based on an appId and the requesting app's Window object.
  */
 
+function invalidFunction(name) {
+	return function() {
+		throw new Error(`Can't initialize ts.io() Hub. ${name}() wasn't passed.`);
+	};
+}
+
 /**
  * The Message Broker AKA The Hub.
  * @param {ChromeWindowFeatures} chrome Special features supplied by the Tradeshift® Chrome™
  */
 export function hub(chrome) {
+	if (hubInstance) {
+		return hubInstance;
+	}
+
+	const {
+		appByWindow = invalidFunction('appByWindow'),
+		windowByApp = invalidFunction('windowByApp')
+	} = chrome;
+
+	/**
+	 * Quickly test that appByWindow & windowByApp work for 'Tradeshift.Chrome'
+	 */
+	{
+		const testChromeWindow = windowByApp(CHROME_APP_ID);
+		const testNotWindow = !(testChromeWindow instanceof Window);
+		const testNotAppId = appByWindow(testChromeWindow) !== CHROME_APP_ID;
+		if (testNotWindow) {
+			throw new Error(
```

Comment:
could be, but I like to have more helpful error messages. I want to have more of these all around the code to help developers instead of them asking us and getting blocked all the time.